src/quick-lint-js/diagnostic-types.h

```diff
@@ -1239,6 +1239,12 @@
       MESSAGE(QLJS_TRANSLATABLE("octal literal may not have decimal"),         \
               characters))                                                     \
                                                                                \
+  QLJS_ERROR_TYPE(                                                             \
+      error_number_literal_will_lose_precision, "E0212",                       \
+      diagnostic_severity::warning, { source_code_span characters; },          \
+      MESSAGE(QLJS_TRANSLATABLE("number literal will lose precision"),         \
+              characters))                                                     \
```

Comment:
Hmm, I think this message could be clearer. How about this:
> integer cannot be represented and will be rounded to {1}

where `{1}` is filled in with the string.

To include the rounded string in the message, you'll need to carefully allocate a string on the heap like we do for `diag_mismatched_jsx_tags` (see `opening_tag_pretty`):
https://github.com/quick-lint/quick-lint-js/blob/b2927b227199d981da7da45ec9a841dc5e77283f/src/parse-expression.cpp#L2751-L2787